remove clones from the header plugin (#3721)
apollographql/router: .changesets/fix_geal_remove_clones_from_headers_plugin.md

### remove clones from the header plugin ([Issue #3068](https://github.com/apollographql/router/issues/3068))

> The list of header operations was cloned for every subgraph query, and this was increasing latency. By storing them in an reference counted structure, we make sure the overhead is minimal.

Author: ```suggestion
The list of header operations was cloned for every subgraph query, and this was increasing latency. We made sure the overhead is minimal by removing those allocations
```
